skimage/segmentation/chanvese.py

```diff
@@ -0,0 +1,240 @@
+import numpy as np
+from scipy.ndimage import distance_transform_edt as distance
+
+
+def _cv_curvature(phi):
+    """ Returns the 'curvature' of a level set 'phi'.
+    """
+    P = np.pad(phi, 1, mode='edge')
+    fy = (P[2:, 1:-1] - P[0:-2, 1:-1]) / 2.0
+    fx = (P[1:-1, 2:] - P[1:-1, 0:-2]) / 2.0
+    fyy = P[2:, 1:-1] + P[0:-2, 1:-1] - 2 * phi
+    fxx = P[1:-1, 2:] + P[1:-1, 0:-2] - 2 * phi
+    fxy = .25*(P[2:, 2:]+P[:-2, :-2]-P[:-2, 2:]-P[2:, :-2])
+    G = np.sqrt(fx**2+fy**2)
+    K = (fxx*fy**2-2*fxy*fx*fy+fyy*fx**2)/(np.power(fx**2+fy**2, 1.5)+1e-10)
+    KG = K*G
+    return KG/np.max(np.abs(KG))
+
+
+def _cv_heavyside(x, eps=1.):
+    """ Returns the result of a regularised heavyside function of the
+    input value(s).
+    """
+    return 0.5*(1.+2./np.pi*np.arctan(x/eps))
+
+
+def _cv_delta(x, eps=1.):
+    """ Returns the result of a regularised dirac function of the
+    input value(s).
+    """
+    return eps/(eps**2+x**2)
+
+
+def _cv_calculate_averages(img, Hphi):
+    """ Returns the average values 'inside' and 'outside'
+    """
+    H = Hphi
+    Hinv = 1.-H
+    Hsum = np.sum(H)
+    Hinvsum = np.sum(Hinv)
+    avg_inside = np.sum(img*H)
+    avg_oustide = np.sum(img*Hinv)
+    if(Hsum != 0):
```

Comment:
Funny to see this in Python :wink:. Please, get rid of parentheses. Here and below.